python/pylibcudf/pylibcudf/interop.pyx

```diff
@@ -315,3 +322,77 @@ def _to_arrow_scalar(cudf_object, metadata=None):
     # Note that metadata for scalars is primarily important for preserving
     # information on nested types since names are otherwise irrelevant.
     return to_arrow(Column.from_scalar(cudf_object, 1), metadata=metadata)[0]
+
+
+cpdef Table from_dlpack(managed_tensor):
+    """
+    Convert a DLPack DLTensor into a cudf table.
+
+    For details, see :cpp:func:`cudf::from_dlpack`
+
+    Parameters
+    ----------
+    managed_tensor : PyCapsule
+        A 1D or 2D column-major (Fortran order) tensor.
+
+    Returns
+    -------
+    Table
+        Table with a copy of the tensor data.
+    """
+    cdef unique_ptr[table] c_result
+    cdef DLManagedTensor* dlpack_tensor = <DLManagedTensor*>PyCapsule_GetPointer(
```

Comment:
We should add some of the safety checks from https://dmlc.github.io/dlpack/latest/python_spec.html#implementation to ensure that we don't try to consume an already deleted capsule or anything like that.

Reply:
Added one check, lmk if you were thinking of any others

Reply:
The other thing to verify is that PyCapsule_GetPointer doesn't return NULL.